src/Service/Player/Url.php

```diff
@@ -148,7 +148,7 @@ public function __toString()
      */
     public function withAutoplay(bool $autoPlay): self
     {
-        if ($this->autoPlay == $autoPlay) {
+        if ($this->autoPlay === $autoPlay) {
```

Comment:
I don't see a test covering this specifically? I would have expected `$this->autoPlay` to somehow be set to a non-boolean.

Reply:
The trouble was that when `$this->autoPlay === NULL`, when you try to set `$url->withAutoplay(FALSE)` this conditional with a double equal would evaluate to `TRUE` and then `$this->autoPlay` would remain as `NULL`. The tests as they were were hiding this bug b/c the tests would build on a single instance of a `Url` object, first testing `$url->withAutoplay(TRUE)`, which would indeed set the value for `$this->autoPlay`, then follow that assertion by setting `$url->withAutoplay(FALSE)`, which would indeed set the value for `$this->autoPlay` to `FALSE`, behaving as expected.

In short, this would work:

```
$url = new \Lullabot\Mpx\Service\Player\Url();
$url = $url->withAutoPlay(TRUE);
$url = $url->withAutoPlay(FALSE);
$url_string = (string) $url;
/// $url_string has &autoPlay=false
```
But this wouldn't work:
```
$url = new \Lullabot\Mpx\Service\Player\Url();
$url = $url->withAutoPlay(FALSE);
$url_string = (string) $url;
/// $url_string is missing &autoPlay=false. sad panda.
```
I changed the tests so that each assertion acts on a new instance.